src/test/java/org/elasticsearch/search/indicesboost/SimpleIndicesBoostSearchTests.java

```diff
@@ -110,4 +100,207 @@ public void testIndicesBoost() throws Exception {
         assertThat(response.getHits().getAt(0).index(), equalTo("test2"));
         assertThat(response.getHits().getAt(1).index(), equalTo("test1"));
     }
+
+    @Test
+    public void testIndicesBoostWithAlias() throws Exception {
+        createIndex("alias_test1", "alias_test2");
+        ensureGreen();
+        client().admin().indices().prepareAliases().addAlias("alias_test1", "alias1").get();
```

Comment:
Hmm, looks like it's not consistent.
IndicesAliasesRequestBuilder's ```addAlias()```  takes index name then alias while IndicesAliasesRequest's ```addAlias()``` takes alias name then index name.